haystack/backends/elasticsearch_backend.py

```diff
@@ -572,12 +572,33 @@ def search(self, query_string, **kwargs):
         if end_offset is not None and end_offset > start_offset:
             search_kwargs["size"] = end_offset - start_offset
 
+        extra_source_kwargs = {}
+        fields = kwargs.get('fields', [])
+
+        if fields:
+            include_fields = []
+            exclude_fields = []
+            for field in fields:
+                if field.startswith('-'):
+                    exclude_fields.append(field.replace('-', ''))
```

Comment:
Is it possible for field names to have '-' in their name elsewhere other than this prefix?  If so, you could do replace('-', '', 1) to make sure it only replaces the first one.

Reply:
I don't think it is possible when using haystack (because python doesn't allow '-' in symbol names) but it's better to be explicit about only replacing the leading one anyway.